### PR TITLE
listen to wallet api events

### PR DESCRIPTION
**Pull request by joshblum**
depends on https://github.com/keybase/client/pull/15368, will rebase once that is in.

listens to events in `stellar1/notify.go` and outputs as follows:

```
$ kb chat api-listen --wallet
Listening for chat notifications. Config: showExploding: false, showLocal: false, subscribeDevChannels: false
Listening for wallet notifications
{"id":"3462a04639f15a82e8546021d1a90e31","fromAssertion":"test1","fromCurrentUser":true,"toUserType":1,"toAssertion":"test13","amount":"1","currency":"USD","amountDescription":"$1.00 USD","status":2}
{"source":"remote","msg":{"id":283,"channel":{"name":"test1,test13","public":false,"members_type":"impteamnative","topic_type":"chat"},"sender":{"uid":"166fb78f0f44d271a2d9065272a67b19","username":"test13","device_id":"b70b55d0127d940cec2733387aa4ed18","device_name":"tmp2"},"sent_at":1547060957,"sent_at_ms":1547060957018,"content":{"type":"sendpayment","send_payment":{"paymentID":"0da0ebad57a8e4a7f9008be44f5c01940d19e4550154f31dc6764d8b655eba5a"}},"prev":null,"unread":false,"channel_mention":"none"},"pagination":{"next":"cd011b","previous":"cd011b","num":1,"last":false}}

{"source":"remote","msg":{"id":285,"channel":{"name":"test1,test13","public":false,"members_type":"impteamnative","topic_type":"chat"},"sender":{"uid":"166fb78f0f44d271a2d9065272a67b19","username":"test13","device_id":"b70b55d0127d940cec2733387aa4ed18","device_name":"tmp2"},"sent_at":1547061053,"sent_at_ms":1547061053079,"content":{"type":"requestpayment","request_payment":{"requestID":"749a67215aedbcf00f41a1e9624e1731","note":"pls"}},"prev":null,"unread":false,"channel_mention":"none"},"pagination":{"next":"cd011d","previous":"cd011d","num":1,"last":false}}
{"source":"remote","msg":{"id":286,"channel":{"name":"test1,test13","public":false,"members_type":"impteamnative","topic_type":"chat"},"sender":{"uid":"1b4f0e9851971998e732078544c96b19","username":"test1","device_id":"6fb1e234cad5e24be2fa84809ef0d518","device_name":"newdev1"},"sent_at":1547061244,"sent_at_ms":1547061244470,"content":{"type":"sendpayment","send_payment":{"paymentID":"a5c779fba89235ed6fd8a2db2fc7c590926e3424a904c4192d4862335d71602d"}},"prev":null,"unread":false,"channel_mention":"none"},"pagination":{"next":"cd011e","previous":"cd011e","num":1,"last":false}}
{"id":"a5c779fba89235ed6fd8a2db2fc7c590926e3424a904c4192d4862335d71602d","txID":"a5c779fba89235ed6fd8a2db2fc7c590926e3424a904c4192d4862335d71602d","time":1547061243539,"statusSimplified":1,"statusDescription":"pending","statusDetail":"","showCancel":false,"amountDescription":"8.1496467 XLM","delta":2,"worth":"$1.00 USD","worthAtSendTime":"","issuerDescription":"","fromType":1,"toType":1,"fromAccountID":"GCMPL74J5OVWIUS6TBPPOBTMOWRIMMMUHL6O4AEJWM4C72AWXCPWEFA6","fromAccountName":"new","fromUsername":"test1","toAccountID":"GDVHP3AP65WE2DRYFBHHSXP4SYIEXH566OPZ3AL5JXFNTAVUX2N5DO3Q","toAccountName":"","toUsername":"test13","toAssertion":"","originalToAssertion":"","note":"pls11","noteErr":"","publicNote":"","publicNoteType":"","externalTxURL":"https://stellar.expert/explorer/public/tx/a5c779fba89235ed6fd8a2db2fc7c590926e3424a904c4192d4862335d71602d"}
{"id":"749a67215aedbcf00f41a1e9624e1731","fromAssertion":"test13","fromCurrentUser":false,"toUserType":1,"toAssertion":"test1","amount":"1","currency":"USD","amountDescription":"$1.00 USD","status":2}
{"id":"a5c779fba89235ed6fd8a2db2fc7c590926e3424a904c4192d4862335d71602d","txID":"a5c779fba89235ed6fd8a2db2fc7c590926e3424a904c4192d4862335d71602d","time":1547061243539,"statusSimplified":3,"statusDescription":"completed","statusDetail":"","showCancel":false,"amountDescription":"8.1496467 XLM","delta":2,"worth":"$1.00 USD","worthAtSendTime":"","issuerDescription":"","fromType":1,"toType":1,"fromAccountID":"GCMPL74J5OVWIUS6TBPPOBTMOWRIMMMUHL6O4AEJWM4C72AWXCPWEFA6","fromAccountName":"new","fromUsername":"test1","toAccountID":"GDVHP3AP65WE2DRYFBHHSXP4SYIEXH566OPZ3AL5JXFNTAVUX2N5DO3Q","toAccountName":"","toUsername":"test13","toAssertion":"","originalToAssertion":"","note":"pls11","noteErr":"","publicNote":"","publicNoteType":"text","externalTxURL":"https://stellar.expert/explorer/public/tx/a5c779fba89235ed6fd8a2db2fc7c590926e3424a904c4192d4862335d71602d"}
```

### CODE REVIEW

File: go/client/chat_api_listen_display.go [hunk 1]
```diff
@@ -0,0 +1,177 @@
+package client
```
Comment (joshblum) [inline go/client/chat_api_listen_display.go]: moved this to a separate file so the stellar logic can be separate

File: go/client/wallet_api_listen_display.go [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+	}
+}
+
+func (d *walletNotificationDisplay) displayPaymentDetails(ctx context.Context, accountID stellar1.AccountID, paymentID stellar1.PaymentID) error {
+	details, err := d.cli.GetPaymentDetailsLocal(ctx, stellar1.GetPaymentDetailsLocalArg{
+		AccountID: &accountID,
+		Id:        paymentID,
+	})
+	if err != nil {
+		return err
```
Comment (zapu) [inline go/client/wallet_api_listen_display.go]: we probably want to get this error out to the user instead of just returning it here

Comment (zapu) [reply go/client/wallet_api_listen_display.go]: probably we could print notifications in similar way to how message notifications are printed - where the returned object is of form `{ msg, error }`. It's not ideal but implementers haven't been complaining

File: go/stellar/global.go [hunk 3]
```diff
@@ -286,7 +286,7 @@ func (s *Stellar) handlePaymentNotification(ctx context.Context, obm gregor.OutO
 	if err = s.refreshPaymentFromNotification(ctx, msg.AccountID, msg.PaymentID); err != nil {
 		return err
 	}
-	s.G().NotifyRouter.HandleWalletPaymentStatusNotification(ctx, msg.AccountID, msg.PaymentID)
+	s.G().NotifyRouter.HandleWalletPaymentNotification(ctx, msg.AccountID, msg.PaymentID)
```
Comment (zapu) [inline go/stellar/global.go]: what does this change do?

Comment (joshblum) [reply go/stellar/global.go]: `HandleWalletPaymentNotification` was never fired, i think this was incorrectly copy pasted during the change from ibms to oobms

File: go/client/wallet_api_listen_display.go (hunk 2 above)
Comment (joshblum) [reply go/client/wallet_api_listen_display.go]: ah great point

File: go/stellar/global.go (hunk 3 above)
Comment (joshblum) [reply go/stellar/global.go]: cc @patrickxb unless this was intentional?

**Comment by joshblum** [comment]
updated:
```
$ kb chat api-listen --wallet
Listening for chat notifications. Config: showExploding: false, showLocal: false, subscribeDevChannels: false
Listening for wallet notifications
{"source":"payment_status","notification":{"id":"516c6a9e71368cb119312ee29831e1da2a5b29155bf394c5d8db90527aaeaa89","txID":"516c6a9e71368cb119312ee29831e1da2a5b29155bf394c5d8db90527aaeaa89","time":1547134773816,"statusSimplified":1,"statusDescription":"pending","statusDetail":"","showCancel":false,"amountDescription":"1 XLM","delta":0,"worth":"","worthAtSendTi}
{"source":"payment_status","notification":{"id":"516c6a9e71368cb119312ee29831e1da2a5b29155bf394c5d8db90527aaeaa89","txID":"516c6a9e71368cb119312ee29831e1da2a5b29155bf394c5d8db90527aaeaa89","time":1547134773816,"statusSimplified":3,"statusDescription":"completed","statusDetail":"","showCancel":false,"amountDescription":"1 XLM","delta":0,"worth":"","worthAtSend}
{"source":"payment_status","notification":{"id":"516c6a9e71368cb119312ee29831e1da2a5b29155bf394c5d8db90527aaeaa89","txID":"516c6a9e71368cb119312ee29831e1da2a5b29155bf394c5d8db90527aaeaa89","time":1547134773816,"statusSimplified":3,"statusDescription":"completed","statusDetail":"","showCancel":false,"amountDescription":"1 XLM","delta":0,"worth":"","worthAtSend}


{"source":"remote","msg":{"id":293,"channel":{"name":"test1,test13","public":false,"members_type":"impteamnative","topic_type":"chat"},"sender":{"uid":"166fb78f0f44d271a2d9065272a67b19","username":"test13","device_id":"b70b55d0127d940cec2733387aa4ed18","device_name":"tmp2"},"sent_at":1547134800,"sent_at_ms":1547134800082,"content":{"type":"requestpayment","re}
{"source":"payment_status","notification":{"id":"a925bd64dba4c8d9e6874e707bc53e7a0803af482cb5747bd3c562eccba1ead8","txID":"a925bd64dba4c8d9e6874e707bc53e7a0803af482cb5747bd3c562eccba1ead8","time":1547134812290,"statusSimplified":1,"statusDescription":"pending","statusDetail":"","showCancel":false,"amountDescription":"96.6430149 XLM","delta":2,"worth":"$11.00 }
{"source":"remote","msg":{"id":294,"channel":{"name":"test1,test13","public":false,"members_type":"impteamnative","topic_type":"chat"},"sender":{"uid":"1b4f0e9851971998e732078544c96b19","username":"test1","device_id":"6fb1e234cad5e24be2fa84809ef0d518","device_name":"newdev1"},"sent_at":1547134812,"sent_at_ms":1547134812639,"content":{"type":"sendpayment","sen}
{"source":"request","notification":{"id":"48a8d733ef7c03e105154df80fb9e631","fromAssertion":"test13","fromCurrentUser":false,"toUserType":1,"toAssertion":"test1","amount":"1","currency":"USD","amountDescription":"$1.00 USD","status":2}}
{"source":"payment_status","notification":{"id":"a925bd64dba4c8d9e6874e707bc53e7a0803af482cb5747bd3c562eccba1ead8","txID":"a925bd64dba4c8d9e6874e707bc53e7a0803af482cb5747bd3c562eccba1ead8","time":1547134812290,"statusSimplified":3,"statusDescription":"completed","statusDetail":"","showCancel":false,"amountDescription":"96.6430149 XLM","delta":2,"worth":"$11.0}
```

File: go/stellar/global.go (hunk 3 above)
Comment (patrickxb) [reply go/stellar/global.go]: I can't remember if it was intentional.  @buoyad do you listen for both of these still?  payment_notification and payment_status_notification.

Comment (buoyad) [reply go/stellar/global.go]: No we don't listen to either of those anymore

**Pull request merged**